main.py

```diff
@@ -127,10 +145,57 @@ def update_db(
     conn.commit()
     conn.close()
 
+# Calculate total cost and tokens for detailed report
+def calculate_cost_and_tokens(result):
+
+    total_embedding_cost = None
+    total_embedding_tokens = None
+    total_llm_cost = None
+    total_llm_tokens = None
+        
+    # Extract 'extraction_result' from the result
+    extraction_result = result.get("extraction_result", [])
+        
+    if not extraction_result:
+        return None, None, None, None
+        
+    extraction_data = extraction_result[0].get("result", "")
+    
+    # If extraction_data is a string, attempt to parse it as JSON
+    if isinstance(extraction_data, str):
+        try:
+            extraction_data = json.loads(extraction_data) if extraction_data else {}
+        except json.JSONDecodeError:
+            logger.warning("Failed to decode JSON for extraction data; defaulting to empty dictionary.")
+            extraction_data = {}
+
+    
+    metadata = extraction_data.get("metadata", None)
+    embedding_llm = metadata.get("embedding") if metadata else None
+    extraction_llm = metadata.get("extraction_llm") if metadata else None
+
+    #Process embedding costs and tokens if embedding_llm list exists and is not empty
+    if embedding_llm and not []:
```

Comment:
@pk-zipstack  we don't need the second part. It is redundant in python. The below condition will make sure embedding_llm is not any of the following None, Empty, False, 0 etc. Make same change for `extraction_llm` as well
```suggestion
    if embedding_llm:
```